samcli/commands/init/__init__.py

```diff
@@ -126,12 +147,20 @@ def do_cli(
     # check for required parameters
     if location or (name and runtime and dependency_manager and app_template):
         # need to turn app_template into a location before we generate
-        extra_context = None
         if app_template:
             templates = InitTemplates(no_interactive, auto_clone)
             location = templates.location_from_app_template(runtime, dependency_manager, app_template)
             no_input = True
-            extra_context = {"project_name": name, "runtime": runtime}
+            default_context = {"project_name": name, "runtime": runtime}
+            if extra_context is None:
+                extra_context = default_context
+            else:
+                merged_context = default_context.copy()
+                extra_context_dict = json.loads(extra_context)
```

Comment:
How are we handling it when users provide invalid JSON?

Reply:
added test.